test/integration/warm-pool/warm_pool_suite_test.go

```diff
@@ -0,0 +1,323 @@
+	netMap := make(map[string]int)
+	fmt.Printf("\nMAX Warm Pool over test duration: \n")
+
+	for i := 0; i < len(resultNetWarm.Data.Result); i++ {
+		node := resultNetWarm.Data.Result[i].Metric.Node
+		var maxArr []int
+		for j := 0; j < len(resultNetWarm.Data.Result[i].Values); j++ {
+			val, _ := strconv.Atoi(resultNetWarm.Data.Result[i].Values[j][1].(string))
+			maxArr = append(maxArr, val)
+			if j == len(resultNetWarm.Data.Result[i].Values)-1 {
+				netMap[node] = val
+			}
+		}
+		fmt.Printf("%v : %v \n", node, slices.Max(maxArr))
+	}
+	fmt.Printf("\nNET Warm Pool over test duration: \n")
+	for k, v := range netMap {
+		fmt.Printf("%v : %v \n", k, v)
+	}
+
+	fmt.Printf("\nNo addresses available error over test duration: \n")
+	for i := 0; i < len(resultNoAddrs.Data.Result); i++ {
+		node := resultNoAddrs.Data.Result[i].Metric.Node
+		var maxArr []int
+		for j := 0; j < len(resultNoAddrs.Data.Result[i].Values); j++ {
+			val := resultNoAddrs.Data.Result[i].Values[j][1].(string)
+			floatVal, err := strconv.ParseFloat(val, 64)
+			if err != nil {
+				Expect(err).ToNot(HaveOccurred())
+			}
+			maxArr = append(maxArr, int(floatVal))
+		}
+		fmt.Printf("%v : %v \n", node, slices.Max(maxArr))
+	}
+
+	fmt.Printf("\nMAX DELTA over test duration: \n")
+	for i := 0; i < len(resultNet.Data.Result); i++ {
+		node := resultNet.Data.Result[i].Metric.Node
+		var maxArr []int
+		for j := 0; j < len(resultNet.Data.Result[i].Values); j++ {
+			val := resultNet.Data.Result[i].Values[j][1].(string)
+			floatVal, err := strconv.ParseFloat(val, 64)
+			if err != nil {
+				Expect(err).ToNot(HaveOccurred())
+			}
+			maxArr = append(maxArr, int(floatVal))
+		}
+		fmt.Printf("%v : %v \n", node, slices.Max(maxArr))
+	}
+}
+
+// Gets the duration in minutes for Prometheus queries
+func strDurationMin(start int64, end int64) string {
+	duration := (end - start) / 60
+	durationMin := strconv.FormatInt(duration, 10)
+	print("TEST DURATION: ", duration)
+	return durationMin
+}
+
+// Random operation, if preventNoChange is 0 this includes no change being a result, otherwise it will add or subtract
+func randOp(replicas int, pods int) (int, string) {
+	if preventNoChange == 0 {
+		op := rand.Intn(3)
+		if op == 0 {
+			return replicas + pods, "adding"
+		}
+		if op == 1 {
+			return replicas - pods, "subtracting"
+		} else {
+			return replicas, "no change"
+		}
+	} else {
+		op := rand.Intn(2)
+		if op == 0 {
+			return replicas + pods, "adding"
+		} else {
+			return replicas - pods, "subtracting"
+		}
+	}
+}
+
+func createCurlPod() {
+	// Run Curl Pod
+	By("Starting Curl Container")
+	curlContainer := manifest.NewCurlContainer().
+		Command([]string{"sleep", "1000"}).Build()
+
+	curlPod := manifest.NewDefaultPodBuilder().
+		Name("curl-pod").
+		Namespace(utils.DefaultTestNamespace).
+		HostNetwork(true).
+		Container(curlContainer).
+		Build()
+
+	testPod, _ := f.K8sResourceManagers.PodManager().
+		CreateAndWaitTillPodCompleted(curlPod)
```

Comment:
How long does this take? I am a bit worries this is waiting all 1000 seconds for the curl container to complete

Reply:
I think I might have been understanding this incorrectly, but I was following existing code in pod_egress_test to create a curl pod and use it to run multiple commands later. However, I am creating the curl pod after the test is completed and not before to not impact the test. I was under the impression that the pod would sleep and be available for any commands in x amount of time, but as running the Prometheus queries doesn't take long the curl pod will be deleted when the command to delete the curl pod runs.  If I don't include some kind of a sleep, I get an error "unable to upgrade connection: container not found." But I think I may be misunderstanding how this works and there is a better solution here. 

tldr; It's never taken 1000 seconds, I do have a time.sleep with changeReplicas mostly to be able to better visualize metrics in the dashboard.

Reply:
All other comments should be resolved but this this one.